gprofiler/main.py

```diff
@@ -851,6 +849,19 @@ def verify_preconditions(args: configargparse.Namespace) -> None:
         print("--log-usage is available only when run as a container!", file=sys.stderr)
         sys.exit(1)
 
+    if args.pids_to_profile is not None:
+        pids_to_profile = []
+        for pid in args.pids_to_profile:
+            try:
+                Process(pid)
+                pids_to_profile.append(pid)
+            except NoSuchProcess:
+                continue
+        setattr(args, "pids_to_profile", pids_to_profile)
+        if len(pids_to_profile) == 0:
+            print("There aren't any alive processes from provided via --pid PID list")
+            sys.exit(1)
```

Comment:
This check itself needs to be in `verify_preconditions`. However, the translation part (PIDs -> Process objects) should not be here - this function is only for verification.

I think the correct place to put it is in `main()`, after the `parse_cmd_args()` call, add a function `pids_to_processes` which modifies `args` to keep only the live processes.